app.py

```diff
@@ -686,155 +700,152 @@ def publisher_snap_measure(snap_name):
-        percentages = []
-        for daily_percent in country_percentages['values']:
-            if daily_percent is not None:
-                percentages.append(daily_percent)
-
-        if len(percentages) > 0:
-            users_by_country[country_code] = (
-                sum(percentages) / len(percentages)
+    max_users = 0.0
+    for country_counts in geodata:
+        country_code = country_counts['name']
+        users_by_country[country_code] = {}
+        counts = []
+        for daily_count in country_counts['values']:
+            if daily_count is not None:
+                counts.append(daily_count)
+
+        if len(counts) > 0:
+            users_by_country[country_code]['number_of_users'] = (
+                sum(counts)
+            )
+            users_by_country[country_code]['percentage_of_users'] = (
+                sum(counts) / len(counts)
             )
         else:
-            users_by_country[country_code] = None
+            users_by_country[country_code]['number_of_users'] = 0
+            users_by_country[country_code]['percentage_of_users'] = 0
+
+        if max_users < users_by_country[country_code]['percentage_of_users']:
+            max_users = users_by_country[country_code]['percentage_of_users']
+
+    calculate_colors(users_by_country, max_users)
 
     # Build up country info for every country
     country_data = {}
     territories_total = 0
     for country in pycountry.countries:
-        number_of_users = randint(0, 20)  # TODO: this is dummy data
-        if number_of_users > 0:
-                territories_total += 1
+        country_info = users_by_country.get(country.alpha_2)
+        number_of_users = 0
+        percentage_of_users = 0
+        color_rgb = [229, 245, 223]
+        if country_info is not None:
+            number_of_users = country_info['number_of_users'] or 0
+            percentage_of_users = country_info['percentage_of_users'] or 0
+            color_rgb = country_info['color_rgb'] or [229, 245, 223]
+
         country_data[country.numeric] = {
             'name': country.name,
             'code': country.alpha_2,
-            'percentage_of_users': users_by_country.get(country.alpha_2),
-            'number_of_users': number_of_users
+            'number_of_users': number_of_users,
+            'percentage_of_users': percentage_of_users,
+            'color_rgb': color_rgb
         }
-    # end of dummy data
+
+        if number_of_users > 0:
+            territories_total += 1
 
     context = {
         # Data direct from details API
         'snap_title': details['title'],
         'package_name': details['package_name'],
-        'metric_period': metric_period_int,
+        'metric_period': metric_period,
 
         # Metrics data
-        'installs_total': sum(installs_values),
-        'installs_metrics': installs_metrics,
-        'active_devices_total': active_devices_total,
+        'installs_total': "{:,}".format(installs_total),
```

Comment:
What is "{:,}" doing?

Reply:
converts 1000 to 1,000 or 10000 to 10,000 @bartaz